src/PublicGoodsDonator.sol

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import {FilAddress} from "shim/FilAddress.sol";
+import {OwnedClaimable} from "shim/OwnedClaimable.sol";
+import {IPreStake} from "src/Interfaces/IPreStake.sol";
+import {IWFIL} from "src/Interfaces/IWFIL.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+/**
+ * @title PublicGoodsDonator
+ * @author GLIF
+ * @notice The PublicGoodsDonator contract is used as a proxy over the Infinity Pool to donate a % of iFIL tokens to public goods
+ */
+contract PublicGoodsDonator is OwnedClaimable {
+
+  using FilAddress for address;
+
+  event Donate(address indexed account, uint256 donationAmount);
+  event WithdrawFunds(address indexed wallet, uint256 amount);
+
+  /// @dev WAD is used to compute the split % to pg wallet
+  uint256 immutable DENOM = 1e18;
+
+  /// @dev modifier to ensure `donationPercentage` is 100% max
+  modifier noOverDonations(uint256 donationPercentage) {
+    if (donationPercentage > DENOM) {
+      revert InvalidParams();
+    }
+    _;
+  }
+
+  IPreStake public immutable preStake;
```

Comment:
Might want to be able to set/upgrade this - probably not necessary

Reply:
Would rather actually just use pause/unpause on it